node/src/Pos/Block/Network/Listeners.hs

```diff
@@ -64,9 +65,10 @@ handleGetBlocks oq = listenerConv oq $ \__ourVerInfo nodeId conv -> do
     whenJust mbMsg $ \mgb@MsgGetBlocks{..} -> do
         logDebug $ sformat ("handleGetBlocks: got request "%build%" from "%build)
             mgb nodeId
-        mHashes <- getHeadersFromToIncl @ssc mgbFrom mgbTo
-        case mHashes of
-            Just hashes -> do
+        -- We fail if we're requested to give more than
+        -- recoveryHeadersMessage headers at once.
+        getHeadersRange @ssc (Just $ recoveryHeadersMessage - 1) mgbFrom mgbTo >>= \case
```

Comment:
AFAIU, currently interface of `getHeadersRange` is that the first argument is the maximal number of headers (though I am not sure and wrote that I don't like the documentation). So I guess you don't need `- 1` here.